### ISSUE TITLE
polish: add inheritable contract with usingOptics client functions

### ISSUE THREAD

**Issue by ltchang2019** [celo-org/optics-monorepo#181]
Inheritable contract to reduce duplicated code between contracts using `usingOptics`. Should have following functionality:
```
- Constructor
    - takes Param usingOptics (address)
    - Calls setUsingOptics
- State var
    - UsingOptics public usingOptics
- Function
    - setUsingOptics
    - enqueueHome
- Modifier
    - onlyReplica 
```

Suggested here: https://github.com/celo-org/optics-monorepo/pull/170#pullrequestreview-619575217

**Comment by prestwich** [comment]
is this related?
 https://github.com/celo-org/optics-monorepo/blob/main/solidity/optics-xapps/contracts/XAppConnectionClient.sol

**Comment by ltchang2019** [comment]
Closed by #343
https://github.com/celo-org/optics-monorepo/blob/main/solidity/optics-xapps/contracts/XAppConnectionClient.sol
